Documentation/project-docs/developer-guide.md

```diff
@@ -16,4 +11,235 @@ The CoreFX repo can be built from a regular, non-admin command prompt. The build
+```
+
+- Building in debug mode for platform x64 targeting OS Linux
+```
+build-managed -debug -buildArch=x64 -os=Linux
+```
+
+- Building only the managed libraries with /flp:v=diag. It doesn't restore packages, builds tests or builds packages.
+```
+build-managed -binaries -verbose
+```
+
+- Building the managed libraries and the tests but don't run the tests.
+```
+build-managed -skiptests
+```
+
+- The following example shows the argument `--`. Everything that is after it is not going to be processed and it is going to be passed as it is.
+Use it to pass extra msbuild properties.
+```
+build-managed -binaries -- /p:WithoutCategories=IgnoreForCI
+```
+
+### Building individual CoreFx DLLs
+
+Under the src directory is a set of directories, each of which represents a particular assembly in CoreFX.
+  
+For example the src\System.Diagnostics.DiagnosticSource directory holds the source code for the System.Diagnostics.DiagnosticSource.dll assembly. Each of these directories includes two projects, one for the DLL being built and one for the tests, both specified by a .builds file.
+
+You can build the DLL for System.Diagnostics.DiagnosticSource.dll by going to the src\System.Diagnostics.DiagnosticsSource\src directory and typing `msbuild System.Diagnostics.DiagnosticSource.builds`. The DLL ends up as  bin\AnyOS.AnyCPU.Debug\System.Diagnostics.DiagnosticSource\System.DiagnosticSource.dll.
+
+You can build the tests for System.Diagnostics.DiagnosticSource.dll by going to 
+src\System.Diagnostics.DiagnosticSource\tests and typing `msbuild System.Diagnostics.DiagnosticSource.Tests.builds`.
+
+There is also a pkg directory for each project, and if you go into it and type `msbuild`, it will build the DLL (if needed)
+and then also build the NuGet package for it. The NuGet package ends up in the bin\packages directory.  
+
+### Building other OSes
+
+By default, building from the root will only build the libraries for the OS you are running on. One can
+build for another OS by specifying `build-managed -FilterToOSGroup=[value]` or build for all by specifying `build-managed -BuildAllOSGroups`.
+
+### Building in Release or Debug
+
+By default, building from the root or within a project will build the libraries in Debug mode. 
+One can build in Debug or Release mode from the root by doing `build -release` or `build -debug` or when building a project by specifying `/p:ConfigurationGroup=[Debug|Release]` after the `msbuild` command.
+
+### Building other Architectures
+
+One can build 32 or 64 bit binaries or for any architecture by specifying in the root `build -buildArch=[value]` or in a project `/p:Platform=[value]` after the `msbuild` command.
+
+
+### Tests
+
+We use the OSS testing framework [xunit](http://xunit.github.io/).
+
+#### Running tests on the command line
+
+By default, the core tests are run as part of the build. Running the tests from
+the command line is as simple as invoking `build.cmd` on windows, and `run-test.sh` on linux and osx.
```

Comment:
The Windows and Linux/Mac commands given here have different behaviors: build.cmd both builds and runs tests, whereas run-test.sh only runs tests.  I think it would be better to give consistent commands for the various OSes—probably one that just runs all of the tests for the repo without building.